sigilrecorder.lic

```diff
@@ -0,0 +1,205 @@
+		
+	end #check_room_data
+	
+	def unstuck_me(target_room)
+		exit_list = Map[target_room].wayto
+		exit_list.each do |rnum, dir|
+		  Flags.add('room-detected', /Room Number: \d+/)
+		  fput("peer #{dir}")
+		  pause
+		  if Flags['room-detected']
+		    fput('retreat')
+		    fput(way)
+			Flags.delete('room-detected')
+			return
+		  end
+		end
+		respond('Cannot find a way out. Exiting script.')
+		write_to_file
+		exit
+	end #unstuck_me
+	
+	
+	def find_sigil? 
+		bput('weave', /You weave back and forth in a dance-like motion/, /You must be closer to use tactical abilities on your opponent/, /There is nothing else to face/)
+		if Flags['danger-room']
+		  add_to_nogo(Room.current.id)
+		  return true
+		end 
+		case line = bput('perceive sigil', @sigil_list, 'Roundtime', /^You recall.*/, /^Having recently been searched/, /You are too distracted/, /Something about/)
+		when /^You recall.*/
+		  respond('Room searched too recently')
+		  return true
+		when /You are too distracted/
+		  add_to_nogo(Room.current.id)
+		  return true
+		when /Something about/
+		  add_to_nogo(Room.current.id)
+		  return true
+		when /^Having recently been searched/
+		  return true
+		when *@sigil_list
+		  @room_sigils.push(line)
+		  respond("Found: #{@room_sigils}")
+		end
+		
+		respond("Array Length: #{@room_sigils.length}")
+		(@room_sigils.length == 2) ? (return false) : find_sigil?
+	end
+	
+	def region_room_list
+		data = Map.list.select do |room|
+		  room.image == Room.current.image
+		end
+
+		roomlist = []
+		data.each do |room|
+		  roomlist.push(room.id)
+		end
+
+		File.write('filtered_map.yaml', roomlist.to_yaml);
+		@walk_list = YAML.load(File.read("filtered_map.yaml"))
+		respond("Loaded #{@walk_list.length} rooms from YAML")
+	end #region_room_list
+	
+	def add_to_nogo(banroom)
+		@data['nogo'].push(banroom)
+		@data['nogo'].uniq
+		respond("Unsuitable room for sigil harvesting. Adding Room #{banroom} to no-go list.")
+		Flags.reset('danger-room')
+		pause 0.2
+	end #add_to_nogo
+	
+	
+	def generate_nogo_list
+	  respond("Pruning walk list to avoid combat. Current list size: #{@walk_list.length}")
+	  nogo_list = []
+	  @avoids = get_data('hunting').hunting_zones.values
+	  @data['nogo'] = (@data['nogo'] + nogo_list).uniq
+	  respond("No-go List: #{@data['nogo'].length}")
+	  @walk_list -= @data['nogo']
+	  respond("Walk list has been pruned. New list size: #{@walk_list.length}")
+	  respond('Starting in 10 seconds...')
+	  pause 10
+	end #generate_nogo_list
+	
+	
+	def write_to_file
+		@room_sigils.each do |sigil|
+		  if @data[@region][sigil][@season] == nil
+		    @data[@region][sigil][@season] = []
+		  end #if
+		  @data[@region][sigil][@season].push(@current_room)
+		  @data[@region][sigil][@season].uniq
+		  #respond("Adding to YAML: #{@data['shard'][sigil][@season]}")
+		next
+		end
+		
+		File.open("./scripts/data/base-sigils.yaml", "w") { |file| file.write(@data.to_yaml) }
+	end #write_to_file
+	
+end #SigilFinder
```

Comment:
Should this be `SigilRecorder`? Incidentally, this is why I'm not a fan of the end comments - they're too easy to get stale.